astropy/modeling/core.py

```diff
@@ -1874,11 +1874,7 @@ def __getattr__(cls, attr):
 
         raise AttributeError(attr)
 
-    def __repr__(cls):
-        if cls._tree is None:
-            # This case is mostly for debugging purposes
-            return cls._format_cls_repr()
-
+    def _get_expr_comp(cls):
```

Comment:
Instead, just call this `_format_components` and have it return the components part--not the list of `keywords` tuples.  Leave the part that builds the list of `keywords` in `__repr__` where it was.  Same in `__str__` below.  In some sense that actually has a little bit *more* code duplication, but uses the interfaces more clearly I think, and will be easier to extend later if needed.

Also move this method lower in the class--I've tried to consistently keep all "internal" methods toward the end of classes in this module.